drivers/gles3/rasterizer_scene_gles3.cpp

```diff
@@ -2730,8 +2734,16 @@ void RasterizerSceneGLES3::_render_list_template(RenderListParameters *p_params,
 			material_data = surf->material_shadow;
 			mesh_surface = surf->surface_shadow;
 		} else {
-			shader = surf->shader;
-			material_data = surf->material;
+			if (get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_OVERDRAW) {
+				material_data = overdraw_material_data_ptr;
+				shader = material_data->shader_data;
+			} else if (get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_LIGHTING) {
```

Comment:
```suggestion
			} else if (unlikely(get_debug_draw_mode() == RS::VIEWPORT_DEBUG_DRAW_LIGHTING)) {
```
Needs to be wrapped in `unlikely()` to help with performance

Reply:
just wondering, what does this do? tried following the macro expansion on my IDE, and it seems like it just expands to the same expression

Reply:
Its a hint to the compiler to tell it that the if statement is most likely to be false. This allows it to emit some special assembly that generally makes the statement faster when it fails.

Without the hint the compiler will make a guess about how often the branch succeeds. If the compiler guesses wrong then it will be slightly slower